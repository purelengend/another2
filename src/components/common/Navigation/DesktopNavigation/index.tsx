import React from 'react'
import { Flex, Button } from '@chakra-ui/react'
import { Link } from 'react-router-dom'
import { NavigationLink } from '@types'

// Desktop Navigation Component
const DesktopNavigation: React.FC<{ links: NavigationLink[] }> = ({
  links,
}) => (
  <Flex as="nav" className="navigation" paddingTop={1.5}>
    <Flex as="ul" className="nav-list" gap={6}>
      {links.map(({ label, href }) => (
        <Flex as="li" className="nav-item" key={label}>
          <Button
            as={Link}
            to={href}
            variant="ghost"
            aria-label={label}
            width="full"
            fontSize="md"
            textTransform="uppercase"
            color="textSecondary"
            border="1px"
            borderColor="transparent"
            borderRadius="full"
            transition="all 0.4s ease-in-out"
            _hover={{
              bgColor: 'transparent',
              borderColor: 'textSecondary',
            }}
          >
            {label}
          </Button>
        </Flex>
      ))}
    </Flex>
  </Flex>
)

export default DesktopNavigation
