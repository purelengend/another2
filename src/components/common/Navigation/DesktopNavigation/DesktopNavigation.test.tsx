import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import DesktopNavigation from '.'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

const mockLinks = [
  { label: 'Home', href: '/home' },
  { label: 'About', href: '/about' },
  { label: 'Contact', href: '/contact' },
]

test('renders DesktopNavigation component with links', () => {
  const { getByText, container } = render(
    <MemoryRouter>
      <DesktopNavigation links={mockLinks} />
    </MemoryRouter>
  )

  // Check if navigation element is rendered
  const navigationElement = container.querySelector('.navigation')
  expect(navigationElement).toBeInTheDocument()

  // Check if links are rendered
  mockLinks.forEach(link => {
    const linkElement = getByText(link.label)
    expect(linkElement).toBeInTheDocument()
    expect(linkElement).toHaveAttribute('href', link.href)
  })

  expect(container).toMatchSnapshot()
})
