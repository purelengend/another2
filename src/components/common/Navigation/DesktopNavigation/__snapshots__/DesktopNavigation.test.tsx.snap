// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DesktopNavigation component with links 1`] = `
<div>
  <nav
    class="navigation css-1cd4dnj"
  >
    <ul
      class="nav-list css-ffmerx"
    >
      <li
        class="nav-item css-k008qs"
      >
        <a
          aria-label="Home"
          class="chakra-button css-1pbf9v6"
          href="/home"
        >
          Home
        </a>
      </li>
      <li
        class="nav-item css-k008qs"
      >
        <a
          aria-label="About"
          class="chakra-button css-1pbf9v6"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        class="nav-item css-k008qs"
      >
        <a
          aria-label="Contact"
          class="chakra-button css-1pbf9v6"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
